src/server.py

```diff
@@ -1,26 +1,182 @@
 """Module to set up the server."""
 import socket
+import email.utils
+import os
+import mimetypes
+import sys
 
 
-def server():
+def server(port=5000):
     """Start server to receive message and echo back."""
     server = socket.socket(socket.AF_INET,
                            socket.SOCK_STREAM,
                            socket.IPPROTO_TCP)
-    address = ("127.0.0.1", 5011)
+    address = ("127.0.0.1", port)
     server.bind(address)
     server.listen(1)
     buffer_length = 8
+    print("Server running on port " + str(port))
     while True:
         try:
             conn, addr = server.accept()
-            message = ""
-            while message[-2:] != "\r\n":
-                message += conn.recv(buffer_length).decode("utf8")
-            conn.sendall(message.encode("utf8"))
+            request = b""
+            response = b""
+            conn.settimeout(5)
+            try:
+                while request[-4:] != b'\r\n\r\n':
```

Comment:
This will break your server for requests that don't use this termination sequence